R/server.R

```diff
@@ -52,7 +52,7 @@ registerClient <- function(client) {
 #' @export
 addResourcePath <- function(prefix, directoryPath) {
   prefix <- prefix[1]
-  if (!grepl('^[a-z][a-z0-9\\-_.]*$', prefix, ignore.case=TRUE, perl=TRUE)) {
+  if (!grepl('^[^\\.][a-z0-9\\-_.]*$', prefix, ignore.case=TRUE, perl=TRUE)) {
```

Comment:
I think this just checks that the first character is not `.`, so other weird characters end up getting through. For example:

```
> grepl('^[^\\.][a-z0-9\\-_.]*$', '#abc', ignore.case=TRUE, perl =TRUE)
[1] TRUE
```

If the goal is to prevent a leading `.`, then I think `^[a-z0-9\\-_][a-z0-9\\-_.]*$` is better:

```
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', '#abc', perl =TRUE)
[1] FALSE
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', '.abc', perl =TRUE)
[1] FALSE
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', 'abc', perl =TRUE)
[1] TRUE
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', '_abc', perl =TRUE)
[1] TRUE
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', '-abc', perl =TRUE)
[1] TRUE
> grepl('^[a-z0-9\\-_][a-z0-9\\-_.]*$', '1abc', perl =TRUE)
[1] TRUE
```